### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by jason-p-pickering**
### Related Issues
- DP-XXX
- etc.

## Reviewer:
- [ ] Tests added/updated & passing.
- [ ] Clean linting.
- [ ] Related issue ticket in Jira/GitHub.
- [ ] Documentation added/updated.
- [ ] Code conforms to style guidelines.
- [ ] Well commented.
- [ ] Updates reflected in NEWS.md.
- [ ] Build check passes.

### CODE REVIEW

File: server.R [hunk 1]
```diff
@@ -31,6 +31,40 @@ if (!file.exists(Sys.getenv("LOG_PATH"))) {
 flog.appender(appender.console(), name = "datapack")
 
 
+######## APP starts
+if (interactive()) {
+  # testing url
+  options(shiny.port = 3123)
```
Comment (JordanBalesBAO) [inline server.R:37]: any reason for this port?

File: server.R [hunk 2]
```diff
@@ -89,30 +123,67 @@ shinyServer(function(input, output, session) {
...
+
+      loginAttempt <- tryCatch({
+      user_input$uuid <- uuid::UUIDgenerate()
+      datimutils::loginToDATIMOAuth(base_url =  Sys.getenv("BASE_URL"),
+                        token = token,
+                        app = oauth_app,
+                        api = oauth_api,
+                        redirect_uri= APP_URL,
+                        scope = oauth_scope,
+                        d2_session_envir = parent.env(environment())
```
Comment (JordanBalesBAO) [inline server.R:168]: brilliant

File: server.R [hunk 3]
```diff
@@ -150,6 +209,11 @@ shinyServer(function(input, output, session) {
   })
 
   observeEvent(input$logout, {
+
+
+    req(input$logout)
+    # Gets you back to the login without the authorization code at top
+    updateQueryString("?",mode="replace",session=session)
```
Comment (JordanBalesBAO) [inline server.R:216]: excellent

File: server.R [hunk 4]
```diff
@@ -184,15 +248,14 @@ shinyServer(function(input, output, session) {
   output$uiLogin <- renderUI({
 
     wellPanel(fluidRow(
-      #img(src = "pepfar.png", align = "center"),
       tags$head(tags$script(HTML(jscode_login))), # enter button functionality for login button
       tags$div(HTML('<center><img src="pepfar.png"></center>')),
-      h4("Welcome to the DataPack Validation App. Please login with your DATIM credentials:")
+      h4("Welcome to the DataPack Validation App. You will be redirected to DATIM to authenticate with your credentials.")
```
Comment (JordanBalesBAO) [inline server.R:253]: good idea

**Review by JordanBalesBAO — APPROVED** [review]
As far as im concerned this is ready to merge once the datimutils pr is approved.

File: server.R (hunk 1 above)
Comment (jason-p-pickering) [reply server.R:37]: No particular reason, it just makes it easier I think when testing locally to know which port you are going to end up on.

**Pull request merged**